src/oceans/components/common/Guide.jsx

```diff
@@ -1,29 +1,82 @@
-import React from 'react'
-import Radium from "radium";
-import Typist from "react-typist";
-
-import {getState, setState} from "@ml/oceans/state";
-import guide from "@ml/oceans/models/guide";
-import soundLibrary from "@ml/oceans/models/soundLibrary";
-import styles from "@ml/oceans/styles";
-import colors from "@ml/oceans/styles/colors";
-import I18n from "@ml/oceans/i18n";
-import {Button} from "@ml/oceans/components/common";
-import arrowDownImage from "@public/images/arrow-down.png";
+import React from 'react';
+import Radium from 'radium';
+import Typist from 'react-typist';
+
+import {getState, setState} from '@ml/oceans/state';
+import guide from '@ml/oceans/models/guide';
+import soundLibrary from '@ml/oceans/models/soundLibrary';
+import styles from '@ml/oceans/styles';
+import colors from '@ml/oceans/styles/colors';
+import I18n from '@ml/oceans/i18n';
+import {Button} from '@ml/oceans/components/common';
+import arrowDownImage from '@public/images/arrow-down.png';
+import {
+  startTextToSpeech,
+  stopTextToSpeech,
+  hasTextToSpeechVoices
+} from '@ml/utils/TextToSpeech';
+
+// A timer used for playing typing sounds.
+let guideTypingTimer = undefined;
+
+// Whether text to speech has ever been successfully
+// started via a user click.
+let textToSpeechStartedViaClick = false;
+
+// The current guide, if any, being played as text
+// to speech.
+let textToSpeechCurrentGuide = undefined;
 
 let UnwrappedGuide = class Guide extends React.Component {
-  onShowing() {
-    clearInterval(getState().guideTypingTimer);
-    setState({guideShowing: true, guideTypingTimer: null});
+  onTypingDone() {
+    clearInterval(guideTypingTimer);
+    setState({guideShowing: true});
+    guideTypingTimer = undefined;
   }
 
-  dismissGuideClick() {
-    const dismissed = guide.dismissCurrentGuide();
-    if (dismissed) {
-      soundLibrary.playSound('other');
-    }
+  onTextToSpeechDone() {
+    setState({guideShowing: true});
+    textToSpeechCurrentGuide = undefined;
   }
 
+  onGuideClick = () => {
+    const state = getState();
+    const currentGuide = guide.getCurrentGuide();
+
+    let textToSpeechStarted = false;
+
+    // Start playing text to speech.
+    if (
+      state.textToSpeechLocale &&
+      hasTextToSpeechVoices() &&
+      !textToSpeechStartedViaClick &&
+      textToSpeechCurrentGuide !== currentGuide &&
+      currentGuide
+    ) {
+      if (
+        startTextToSpeech(
+          currentGuide.textFn(getState()),
+          state.textToSpeechLocale,
+          this.onTextToSpeechDone
+        )
+      ) {
+        textToSpeechCurrentGuide = currentGuide;
+        textToSpeechStartedViaClick = true;
+        textToSpeechStarted = true;
+      }
+    }
+
+    if (!textToSpeechStarted) {
+      const dismissed = guide.dismissCurrentGuide();
```

Comment:
nit: I think a couple of comments could make this clearer. Maybe one here that says something like `Handle guide dismissal if we were unable to initiate text-to-speech (due to missing voices, etc.)` and one on line 63 to the effect of `Registers text-to-speech settings for the current guide if successful`.

Reply:
I refactored this code quite a bit, with a centralized `updateTextToSpeech` function that has a bunch of comments.  Hopefully that helps.